### ISSUE TITLE
Limit JsAssoc to cstring and int

### ISSUE THREAD

**Issue by alehander42** [nim-lang/Nim#8601]
JsAssoc can accept various K types for a key, which is something that I happily used. 
However this usually leads to subtle bugs: JS converts all keys to strings, so 

1  conversion to string might be not always unique for some types(e.g. objects)
2 when you iterate, the key is actually a cstring, which means the Nim type system leads you to a probable error in this case

e.g. if you have int, int

```
var b = JsAssoc[int, cstring]{4: cstring"e"}
for a, _ in b:
  # a
```

a **can easily seem** like an int, but it's a cstring, a lot of operations on it will work in the same way even if it's cstring, but sooner or later this leads to a weird bug

Basically we need to limit K to types that are easily `<=>` with cstring, like int and make special `pairs` for those types, so we automatically insert a conversion to `K`. (the `V` value type should remain the same)

Thoughts? Should I make a PR?

**Comment by Araq** [comment]
> Thoughts? Should I make a PR?

Please!

**Comment by alehander42** [comment]
this was also merged
